gen/cheader.tmpl

```diff
@@ -37,7 +37,17 @@
 #define WGPU_NULLABLE
 #endif{{"\n" -}}
 
-{{if eq .Name "webgpu"}}
+{{- if ne .Name "webgpu"}}
+#if !defined(__WGPU_EXTEND_ENUM)
+#ifdef __cplusplus
+#define __WGPU_EXTEND_ENUM(E, N, V) static const E N = E(V)
+#else
+#define __WGPU_EXTEND_ENUM(E, N, V) static const E N = (E)(V)
```

Comment:
I wonder if we need a `WGPU_ENUM_EXTENDED_VALUE_ATTRIBUTE` here. I'm not really clear on when Swift needs `NS_REFINED_FOR_SWIFT` exactly, but it seems like it probably would need one here in order to refine our enums into Swift enums?

Reply:
(This does not block landing the PR. I don't think #179 is fully resolved right now anyway.)

Reply:
Let's just defer this to when someone asks for it.